os/mm/mm_heap/mm_getsize.c

```diff
@@ -0,0 +1,90 @@
+/****************************************************************************
+ *
+ * Copyright 2022 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <debug.h>
+#include <sys/types.h>
+#include <tinyara/mm/mm.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+static size_t mm_find_largest_freesize(struct mm_heap_s *heap)
+{
+	size_t largest_size = 0;
+	struct mm_freenode_s *fnode;
+	int i = 0;
+
+	/* Free nodes are sorted in a descending order,
+	 * so the first node in each nodelist is the largest within its nodelist.
+	 */
+	for (i = 0; i < MM_NNODES; ++i) {
+		fnode = heap->mm_nodelist[i].flink;
+		if (fnode && largest_size < fnode->size) {
+			largest_size = fnode->size;
+		}
+	}
+	return largest_size;
+}
+
+/****************************************************************************
+ * Name: mm_get_largest_freenode_size
+ *
+ * Description:
+ *   returns a largest node size in the heap
+ ****************************************************************************/
+
+size_t mm_get_largest_freenode_size(void *address)
```

Comment:
`mm_get_largest_...` and `mm_find_largest_...` looks almost the same operation. I feel `mm_find` returns node and `mm_get` returns the size. But it doesn't.
How about `mm_get_heap_largest_free_...` and `mm_get_space_largest_free...`?
I name that `space` is kernel or user, `heap` is each heap.
Please let me know your opinion.

Reply:
`mm_find` also returns the size. 
What I meant, `mm_get` returns the size to user(because it is public API), `mm_find` is internal function to find the size.
Many other APIs are `mm_get_xxx` like `mm_get_heap`, so I named like this.
I understand what you are worried, but it is not to get space or heap.

Reply:
> mm_find also returns the size.

This is what I think a problem. `find` returns the free size, but `get` also returns the free size. Then both are the same in naming point of view.

> What I meant, mm_get returns the size to user(because it is public API), mm_find is internal function to find the size.

Then, please represent it in the name. ie. _mm_get_.... and mm_get_...
Because both returns free size (same operation).

> I understand what you are worried, but it is not to get space or heap.

`mm_find` only returns the value of one heap even that space has several heap.
But `mm_get` checks entire heap and returns largest.
So I think `mm_find` could be individual heap checking, and `mm_get` could be entire heap checking.

Reply:
I changed `mm_find_largest_freesize` to `mm_find_largest_freesize_from_specific_heap`.

Reply:
It returns size as well. Why is this `find`? not `get`?  In the same module. naming policy should be the same.

Reply:
Modified. Thanks.

Reply:
I think both find or both get is correct.

Reply:
Changed to get for both.